lib/exportPlatform.js

```diff
@@ -43,6 +43,72 @@ function transformPropertyForGroup(property, options) {
   return to_ret;
 }
 
+/**
+ * Overrides filePath with local context and adds 'appliedFrom'
+ *
+ * @context
+ * // source
+ * {
+ *   "external": {
+ *     "ref": {
+ *       "key1": { "value": "x" },
+ *       "key2": { "value": "y" },
+ *     }
+ *   }
+ * }
+ * // target
+ * {
+ *   "myComponent": {
+ *     "key1": { "value": "q" },
+ *     "@apply": { "value" : "{external.ref}" },
+ *     "key2": { "value": "z" },
+ *   }
+ * }
+ *
+ * @param {object} resolvedObj the result of refObj: "external.ref"
+ * @param {object} refObj the object that '@apply' points to: `{ "value" : "{external.ref}" }`
+ * @param {string} unresolvedValue
+ * @param {object|null} obj the object that wraps '@apply': "myComponent"
+ */
+function adoptLocalContextResolvedObject(
+  resolvedObj,
+  refObj,
+  unresolvedValue,
+  obj
+) {
+  const result = {};
+
+  const keysOriginalObj = obj ? Object.keys(obj) : null;
+  for (const key in resolvedObj) {
+    if (key.startsWith("@")) {
+      continue; // skip decorators
+    }
+    const shouldOverride = keysOriginalObj
+      ? keysOriginalObj.indexOf("@apply") > keysOriginalObj.indexOf(key)
+      : true;
+    if (shouldOverride) {
+      const appliedFrom = `${unresolvedValue}.${key}`;
+      let toBeMerged = resolvedObj[key];
+      if (_.isPlainObject(resolvedObj[key]) && !("value" in resolvedObj[key])) {
+        // eslint-disable-next-line no-empty-pattern
+        for (const {} in resolvedObj[key]) {
```

Comment:
what happens here?

Reply:
We call `adoptLocalContextResolvedObject` for all `nextLevelKeys`, only not use the term itself.